File: 0137-single-number-ii/0137-single-number-ii.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        unsigned int ans = 0;

        for (int i = 0 ; i < 32 ; i++){

            int bit_sum = 0;
            for(int num : nums){
                bit_sum += (num >> i) & 1;
            }

            if (bit_sum %3 != 0){  
                ans |= (1U << i);
            }
        }
        return (int)ans;
    }
};

File: 0137-single-number-ii/0137-single-number-ii.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        ans = 0

        for i in range(32):
            bit_sum = 0
            for num in nums :
                bit_sum += (num >> i) & 1
            
            if bit_sum % 3 != 0:

                if i == 31 :
                    ans -= ( 1 << i)
                else :
                    ans |= (1 << i)
        
        return ans